mne/io/ant/ant.py

```diff
@@ -132,14 +140,34 @@ def __init__(
         info = create_info(
             ch_names, sfreq=cnt.get_sample_frequency(), ch_types=ch_types
         )
+        info.set_meas_date(read_meas_date(cnt))
+        make, model, serial, site = read_device_info(cnt)
+        info["device_info"] = dict(type=make, model=model, serial=serial, site=site)
+        his_id, name, sex, birthday = read_subject_info(cnt)
+        info["subject_info"] = dict(
+            his_id=his_id, first_name=name, sex=sex, birthday=birthday
+        )
         if bipolars is not None:
             with info._unlock():
                 for idx in bipolars_idx:
                     info["chs"][idx]["coil_type"] = FIFF.FIFFV_COIL_EEG_BIPOLAR
-        # read and scale data array
-        data = read_data(cnt)
-        _scale_data(data, ch_units)
-        super().__init__(info, preload=data, filenames=[fname], verbose=verbose)
+        first_samps = np.array((0,))
+        last_samps = (cnt.get_sample_count() - 1,)
+        raw_extras = {
+            "orig_nchan": cnt.get_channel_count(),
+            "orig_ch_units": ch_units,
+            "first_samples": np.array(first_samps),
+            "last_samples": np.array(last_samps),
+        }
+        super().__init__(
+            info,
+            preload=preload,
+            first_samps=first_samps,
+            last_samps=last_samps,
+            filenames=[fname],
+            verbose=verbose,
+            raw_extras=[raw_extras],
+        )
```

Comment:
Is there anything we have to be careful about when `preload=False/True` and we then load `Annotations` *after* the call to `super().__init__(...)`?

Reply:
To the best of my understanding from the `BaseRaw` implementation, as long as the correct `first_samps`, and `last_samps` is passed in the `super(). __init__()` call, `Annotations` reading should be good.

Reply:
Added tests that confirm this ;) Great!

Reply:
Thanks a lot!